interfaces/system_key.go

```diff
@@ -378,11 +384,80 @@ func RemoveSystemKey() error {
 }
 
 func MockSystemKey(s string) func() {
-	var sk systemKey
-	err := json.Unmarshal([]byte(s), &sk)
+	sk, err := SystemKeyFromString(s)
 	if err != nil {
 		panic(err)
 	}
-	mockedSystemKey = &sk
+	mockedSystemKey = sk.(*systemKey)
 	return func() { mockedSystemKey = nil }
 }
+
+type SystemKeyMismatchAction int
+
+const (
+	SystemKeyMismatchActionUndefined SystemKeyMismatchAction = iota
+	SystemKeyMismatchActionNone
+	SystemKeyMismatchActionRegenerateProfiles
+)
+
+func (s SystemKeyMismatchAction) String() string {
+	switch s {
+	case SystemKeyMismatchActionNone:
+		return "none"
+	case SystemKeyMismatchActionRegenerateProfiles:
+		return "regenerate-profiles"
+	default:
+		return fmt.Sprintf("<undefined %d>", int(s))
```

Comment:
I'd use something slightly different to make it clear what the value represents.

```suggestion
		return fmt.Sprintf("SystemKeyMismatchAction(%d)", int(s))
```

Reply:
changed